pkg/kubelet/container/runtime_cache_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
```

Comment:
Yes, the tests here do not directly reflect a single failure case, but I think they are testing the more fundamental problem, which is the cache could be updated to older pods. It doesn't have to be related to the specific pod worker case you mentioned.

As previously discussed in this PR, only the updatingThread could potentially cause the false updates if no safeguard is implemented. Given that, the test you want would be like this:
 1. pod_worker1 creates containers A
 2. pod_worker1 calls `ForceUpdateIfOlder`
 3. Dial back the clock in the cache. Set fakeDocker container list to not include A. Call `GetPods` to trigger an update. Make sure the updating thread tries to update the cache but failed(?). 
 4. Restore the clock in the cache. Call `GetPods` to confirm that container A exists.

This doesn't see clear or cleaner to me...